package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,187 @@
+        # TODO default auxnames; allow to name each column + pick which to keep
+
+        self.names = auxnames
+        self.represent_ts_as = represent_ts_as
+        self.cutoff = cutoff ## UNITS?
+        self.time_first_col = time_first_col  ## what if time in another column?
+
+        self.ts_data = None
+        self.ts_rep = None
+
+        self.step = 0
+        self._read_next_step()
+
+        # set initial time; default to 0
+        if time_first_col:
+            init_t = self.time
+        elif initial_time:
+            init_t = intial_time
+        else:
+            init_t = 0 # TODO - warn using default?
+        self.initial_time = init_t
+
+        # set dt (assuming constant!); default to 1 ps.
+        if time_first_col:
+            self._read_next_step()
+            dt = self.time - self.initial_time
+            self.go_to_first_step()
+        elif dt:
+            self.dt = dt
+        else:
+            init_t = 1 # TODO - warn using default; (check units!)
+        self.dt = dt
+        
+
+    def next(self):
+        """ Move to next step of data """
+        return self._read_next_step()
+
+    def __next__(self):
+        """ Move to next step of data """
+        return self.next()
+
+    def __iter__(self):
+        self._restart()
+        return self
+
+    def _restart(self):
+        """ Reset back to start; calling next should read in first step """
+        # Overwrite when reading from file to seek to start of file
+        self.step = 0
+                
+    def go_to_first_step(self):
+        """ Return to and read first step """
+        ## May need to overwrite, depending e.g. how header dealt with
+        ## Could also use go_to_step(1) ...
+        self._restart()
+        self._read_next_step()
+                
+    def _read_next_step(self):  
+        # Define in each auxiliary reader
+        raise NotImplementedError(
+            "BUG: Override _read_next_timestep() in auxilairy reader!")
+
+    def read_next_ts(self, ts):
+        """ Read and record data from steps closest to *ts*, then 
+        calculate representative value for ts """
+        # Make sure auxiliary and trajectory are still aligned
+        if not self.first_in_ts(ts):
+            return self.go_to_ts(ts)
+        self.reset_ts()
+        while self.step_in_ts(ts):
+            self.add_step_to_ts(ts.time)
+            try:
+                self._read_next_step()
+            except StopIteration:
+                break
+        self.ts_rep = self.calc_representative()
+        ts.aux.__dict__[self.names] = self.ts_rep
+        return ts
+        ## currently means that after reading in a timestep, ts_data and
+        ## ts_diffs correspond to that ts but the current step/step_data of 
+        ## the auxreader is the first step 'belonging' of the next ts...
+
+    def first_in_ts(self, ts):
+        """ Check if current step is first step 'belonging' to *ts*
+        Assumes auxiliary *dt* is constant! """
```

Comment:
Instead of assuming that `self.dt` is constant, you could read the previous and the next auxiliary value to know the current `dt`. Might be expensive, though.